Example project is incompatible with Maven 3.6

When running the example project using Maven 3.6, the thorntail plugin fails with somewhat mysterious error message.

The failure is caused by thorntail-maven-plugin:2.2.0.Final being incompatible with Maven 3.6. When Maven 3.5 is used, the example works.

c.f. https://stackoverflow.com/questions/53507591/thorntail-examples-api-incompatibility, and https://issues.redhat.com/browse/THORN-2229

```
[ERROR] Failed to execute goal io.thorntail:thorntail-maven-plugin:2.2.0.Final-redhat-00022:package (default) on project resource-adapter-thorntail-example: Execution default of goal io.thorntail:thorntail-maven-plugin:2.2.0.Final-redhat-00022:package failed: An API incompatibility was encountered while executing io.thorntail:thorntail-maven-plugin:2.2.0.Final-redhat-00022:package: java.lang.AbstractMethodError: null
[ERROR] -----------------------------------------------------
[ERROR] realm =    plugin>io.thorntail:thorntail-maven-plugin:2.2.0.Final-redhat-00022
[ERROR] strategy = org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy
[ERROR] urls[0] = file:/home/jdanek/.m2/repository/io/thorntail/thorntail-maven-plugin/2.2.0.Final-redhat-00022/thorntail-maven-plugin-2.2.0.Final-redhat-00022.jar
[ERROR] urls[1] = file:/home/jdanek/.m2/repository/io/thorntail/fraction-metadata/2.2.0.Final-redhat-00022/fraction-metadata-2.2.0.Final-redhat-00022.jar
[ERROR] urls[2] = file:/home/jdanek/.m2/repository/io/thorntail/meta-spi/2.2.0.Final-redhat-00022/meta-spi-2.2.0.Final-redhat-00022.jar
[ERROR] urls[3] = file:/home/jdanek/.m2/repository/org/jboss/shrinkwrap/descriptors/shrinkwrap-descriptors-impl-javaee/2.0.0.redhat-122
```

--- a/resource-adapter-thorntail-example/pom.xml
+++ b/resource-adapter-thorntail-example/pom.xml
@@ -19,7 +19,7 @@
     <maven.compiler.target>1.8</maven.compiler.target>
     <version.javax.json>1.0.4</version.javax.json>
     <version.junit>4.12</version.junit>
-    <version.thorntail>2.2.0.Final</version.thorntail>
+    <version.thorntail>2.3.0.Final</version.thorntail>
     <failOnMissingWebXml>false</failOnMissingWebXml>
   </properties>
 

